core/new-gui/src/app/workspace/component/operator-panel/operator-panel.component.spec.ts

```diff
@@ -22,4 +35,47 @@ describe('OperatorPanelComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('getGroupNamesSorted test 1', () => {
+    const groups = MOCK_OPERATOR_GROUPS;
+
+    const result = c.getGroupNamesSorted(groups);
+
+    expect(result).toEqual(['Source', 'Analysis', 'View Results']);
+
+  });
+
+  it('getGroupNamesSorted test 2', () => {
+    const groups: GroupInfo[] = [
+      { groupName: 'group_1', groupOrder: 1 },
+      { groupName: 'group_2', groupOrder: 100 }
+    ];
+
+    const result = c.getGroupNamesSorted(groups);
+
+    expect(result).toEqual(['group_1', 'group_2']);
+
+  });
+
+  it('getOperatorGroupMap test 1', () => {
+    const opMetadata = MOCK_OPERATOR_METADATA;
+
+    const result = c.getOperatorGroupMap(opMetadata);
+
+    const expectedResult = new Map<string, OperatorSchema[]>();
+    expectedResult.set('Source', [opMetadata.operators[0]]);
+    expectedResult.set('Analysis', [opMetadata.operators[1]]);
+    expectedResult.set('View Results', [opMetadata.operators[2]]);
+
+    expect(result).toEqual(expectedResult);
+
+  });
+
+  it('should receive operator metadata from service', () => {
+    // if the length of our schema list is equal to the length of mock data
+    // we assume the mock data has been received
+    expect(component.operatorSchemaList.length).toEqual(MOCK_OPERATOR_SCHEMA_LIST.length);
```

Comment:
F2FQ: What is component here?

Reply:
component is declared above, and initialized in the `beforeEach` function, which is called every time before each test